core/types/block.go

```diff
@@ -33,25 +33,53 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
+var (
+	EmptyRootHash  = DeriveSha(Transactions{})
+	EmptyUncleHash = CalcUncleHash(nil)
+)
+
+var (
+	errMissingHeaderMixDigest = errors.New("missing mixHash in JSON block header")
+	errMissingHeaderFields    = errors.New("missing required JSON block header fields")
+	errBadNonceSize           = errors.New("invalid block nonce size, want 8 bytes")
+)
```

Comment:
This would just increase the API surface. If a block is bad, the program doesn't have any meaningful thing to do with it. The user can look at the error message / logs and figure out what went wrong, but unless it's an actionable thing from a program's perspective, it doesn't really make sense to export it.